### PR TITLE
feat: Add context focus extension for convenient keyboard handling

### PR DESCRIPTION
**Pull request by yusufkecer**

### CODE REVIEW

File: lib/src/private/sub_extension/context/context_focus_extension.dart [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import 'package:flutter/material.dart';
+
+/// Extension on [BuildContext] to provide convenient methods for closing the keyboard.
+/// Example usage:
+/// context.unfocus;
+
+extension FocusExtension on BuildContext {
```
Comment (VB10) [inline lib/src/private/sub_extension/context/context_focus_extension.dart:7]: the usage of 'kartal' is not acceptable. You should add it in the context_general extension, and then you need to call context.general.unFocus().

File: lib/src/private/sub_extension/context/context_focus_extension.dart [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  /// - `focusScope`: Returns the [FocusNode] of the current [BuildContext].
+  FocusNode get focusScope => FocusScope.of(this);
+
+  /// - `unfocus`: Unfocuses the current [BuildContext]'s [FocusScope].
+  void get unfocus => focusScope.unfocus();
```
Comment (VB10) [inline lib/src/private/sub_extension/context/context_focus_extension.dart:12]: You should remove the getter because it is a function. Just call unFocus()

**Review by VB10 — CHANGES_REQUESTED** [review]
please fix my comments, and also add unit tests with documentation in the readme

**Comment by yusufkecer** [comment]
rebased and ready

**Review by VB10 — APPROVED** [review]
lgtm <3

**Pull request merged**